components/agenda/src/Agenda.svelte

```diff
@@ -248,29 +248,26 @@
   // #region Data Flow
 
   // Accept either comma-separated string, or array.
-  $: calendarIDs = (() => {
+  let calendarIDs: string[] = [];
+  $: (calendarIDs = setCalendarIDs()), internalProps, calendar_id;
+  function setCalendarIDs() {
     let IDList = internalProps.calendar_ids;
-    if (
-      typeof internalProps.calendar_ids === "string" &&
-      internalProps.calendar_ids.length
-    ) {
-      IDList = IDList.split(",").map((id: string) => id.trim());
+    if (typeof IDList === "string" && IDList.length) {
+      return IDList.split(",").map((id: string) => id.trim());
     } else if (calendar_id) {
-      IDList = [calendar_id];
+      return [calendar_id];
     }
-    return IDList;
-  })();
+    return [];
```

Comment:
it was being set to "" when `internalProps.calendar_ids` doesn't have any, causing type error. 

Fixed that and named this function for readability